pkg/cloud/azure/services/virtualmachines/virtualmachines_test.go

```diff
@@ -92,6 +92,23 @@ func TestDeriveVirtualMachineParameters(t *testing.T) {
 				g.Expect(vm.SecurityProfile.EncryptionAtHost).To(BeNil())
 			},
 		},
+		{
+			name:       "Non-ThirdParty Marketplace Image",
+			updateSpec: nil,
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.Plan).To(BeNil())
+			},
+		},
+		{
+			name: "ThirdParty Marketplace Image",
+			updateSpec: func(vmSpec *Spec) {
+				vmSpec.Image.ThirdPartyImage = true
+			},
+			validate: func(g *WithT, vm *compute.VirtualMachine) {
+				g.Expect(vm.Plan).ToNot(BeNil())
+
+			},
+		},
```

Comment:
Should we add a ShareImageGallery test as well?

Reply:
Just removed support for Image Galleries from the API. Apparently we are only doing marketplace images.